core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/NativeImageResourcePatternsBuildItem.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.deployment.builditem.nativeimage;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.deployment.util.GlobUtil;
+
+/**
+ * A build item that indicates that a set of resource paths defined by regular expression patterns or globs should be
+ * included in the native image. This uses native-image's {@code -H:IncludeResources} under the hood.
```

Comment:
I thought that the purpose of this PR is to avoid `-H:IncludeResources` completely?

Reply:
Haha, true, I have added the comment some time ago, when the migration away from the CLI was not planned yet. I should change the wording. I should say something like "it is using the same mechanism as native-image's -H:IncludeResources command line option".